### PR TITLE
[SPIR-V][DOC] Update SPV_INTEL_joint_matrix extension

### PR DESCRIPTION
**Pull request by MrSidims**
Signed-off-by: Sidorov, Dmitry <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
FYI @dkhaldi @arnamoy10 . This should help with `get_coord` implementation in SYCL headers.

File: sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc [hunk 1]
```diff
@@ -136,19 +181,70 @@ is spread across multiple invocations.
...
-| 0 | *ColumnMajor*            |  *{capability_name}*
-| 1 | *RowMajor*               |  *{capability_name}*
-| 2 | *PackedA* +
-Suitable for VNNI instructions | *{capability_name}*
-| 3 | *PackedB* +
-Suitable for VNNI instructions | *{capability_name}*
+| 0 | *RowMajor*               |  *{main_capability_name}*
+| 1 | *ColumnMajor*            |  *{main_capability_name}*
+| 2 | *Packed* +
+Suitable for Vector Neural Network Instructions (VNNI) used in Intel AMX
```
Comment (masahi) [inline sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc:197]: Seems odd to say "VNNI instructions are used by AMX and XMX". Maybe "VNNI instruction format"?

File: sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc [hunk 2]
```diff
@@ -369,40 +574,49 @@ result of a constant instruction with scalar 'integer type'. +
...
 |=====
 
-[cols="1,1,6*3",width="100%"]
+[cols="1,1,5*3",width="100%"]
 |=====
-7+|[[OpJointMatrixSUMadINTEL]]*OpJointMatrixSUMadINTEL* +
+6+|[[OpJointMatrixSUMadINTEL]]*OpJointMatrixSUMadINTEL* +
  +
 Multiply matrix 'A' by matrix 'B' and add matrix 'C' to the result of the
 multiplication: `A*B+C`. Here 'A' is a `M x K` matrix, 'B' is a `K x N`
```
Comment (vmaksimo) [inline sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc:584]: Nit: maybre
```suggestion
multiplication: `A x B + C`. Here 'A' is a `M x K` matrix, 'B' is a `K x N`
```

File: sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc [hunk 3]
```diff
@@ -412,40 +626,48 @@ result of a constant instruction with scalar 'integer type'. +
...
 |=====
 
-[cols="1,1,6*3",width="100%"]
+[cols="1,1,5*3",width="100%"]
 |=====
-7+|[[OpJointMatrixUSMadINTEL]]*OpJointMatrixUSMadINTEL* +
+6+|[[OpJointMatrixUSMadINTEL]]*OpJointMatrixUSMadINTEL* +
  +
 Multiply matrix 'A' by matrix 'B' and add matrix 'C' to the result of the
 multiplication: `A*B+C`. Here 'A' is a `M x K` matrix, 'B' is a `K x N`
```
Comment (vmaksimo) [inline sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc:636]:
```suggestion
multiplication: `A x B + C`. Here 'A' is a `M x K` matrix, 'B' is a `K x N`
```

File: sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc [hunk 4]
```diff
@@ -455,40 +677,48 @@ result of a constant instruction with scalar 'integer type'. +
...
 |=====
 
-[cols="1,1,6*3",width="100%"]
+[cols="1,1,5*3",width="100%"]
 |=====
-7+|[[OpJointMatrixUUMadINTEL]]*OpJointMatrixUUMadINTEL* +
+6+|[[OpJointMatrixUUMadINTEL]]*OpJointMatrixUUMadINTEL* +
  +
 Multiply matrix 'A' by matrix 'B' and add matrix 'C' to the result of the
 multiplication: `A*B+C`. Here 'A' is a `M x K` matrix, 'B' is a `K x N`
```
Comment (vmaksimo) [inline sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc:687]:
```suggestion
multiplication: `A x B + C`. Here 'A' is a `M x K` matrix, 'B' is a `K x N`
```

File: sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc (hunk 1 above)
Comment (MrSidims) [reply sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc]: Applied, thanks!

**Comment by MrSidims** [comment]
@dpcpp-spirv-doc-reviewers should we merge this? I keep getting questions about where to find debug info spec and open pull request doesn't give enough trust :(

**Review by bashbaug — APPROVED** [review]
This extension still has the "preview extension disclaimer" so I'm fine merging it as-is.  Thanks!

**Pull request merged**